tmt/steps/report/__init__.py

```diff
@@ -79,14 +83,18 @@ class ReportPlugin(tmt.steps.Plugin):
     how = 'display'
 
     # List of all supported methods aggregated from all plugins
-    _supported_methods = []
+    _supported_methods: List[tmt.steps.Method] = []
 
     @classmethod
-    def base_command(cls, method_class=None, usage=None):
+    def base_command(
+            cls,
+            method_class: Optional[Any] = None,
+            usage: Optional[str] = None) -> click.Command:
         """ Create base click command (common for all report plugins) """
 
         # Prepare general usage message for the step
         if method_class:
+            assert isinstance(usage, str)
```

Comment:
What about making the `usage` parameter mandatory instead? Seems more straightforward and it always needs to be set anyway.

Reply:
Is this true for all `base_command()` methods? To me it looks like it should be done in its own patch separately from typing annotations since it is changing how the method is used.

Reply:
Yes, `usage` has to be provided for all commands. Probably not worth of a separate pull request, covered by 809aed5.